agent/xds/clusters.go

```diff
@@ -849,6 +849,31 @@ func (s *ResourceGenerator) configIngressUpstreamCluster(c *envoy_cluster_v3.Clu
 	if threshold != nil {
 		c.CircuitBreakers.Thresholds = []*envoy_cluster_v3.CircuitBreakers_Thresholds{threshold}
 	}
+
+	// Configure the default outlier detector for upstream service
+	outlierDetection := ToOutlierDetection(cfgSnap.IngressGateway.Defaults.PassiveHealthCheck)
+
+	// override the default outlier detection value
+	if svc.PassiveHealthCheck != nil {
```

Comment:
I wonder if it would simplify things to either update `ToOutlierDetection` or make a new function that takes a variable number of `*structs.PassiveHealthCheck`s and `peeredService`. If we do that, I believe we can share this logic everywhere in this file.

Reply:
Done. Add an `override` parameter to `ToOutlierDetection`.